docs/pktoken.md

```diff
@@ -0,0 +1,702 @@
+# A Guide to PK Tokens
+
+OpenPubkey works by binding an public key to an identity described in an ID Token. To provide verifiers with all the needed information to verify this binding between identity and public key we include additional information by extending the ID Token with additional signatures. This is possible because ID Tokens are JSON Web Signatures (JWS) and JWS can support more than one signature. We call an ID Token extended in this fashion a PK Token (Public Key Token).
+
+In this document we provide needed background on JSON Web Signatures (JWS), ID Tokens and PK Tokens. Then we describe how our different types of PK Tokens work using examples. In the next section we provide our compact serialization format. Finally we provide a short guide to using the patterns we developed for generic multi-signature JWS.
+
+- [A Guide to PK Tokens](#a-guide-to-pk-tokens)
+  - [JSON Web Signatures (JWS) and JSON Web Tokens (JWT)](#json-web-signatures-jws-and-json-web-tokens-jwt)
+  - [PK Tokens](#pk-tokens)
+    - [Protected Header Claims](#protected-header-claims)
+      - [Required Claims](#required-claims)
+      - [Custom Claims](#custom-claims)
+    - [Signature Type (typ)](#signature-type-typ)
+    - [Commitment mechanism](#commitment-mechanism)
+    - [Types of Signatures in a PK Token](#types-of-signatures-in-a-pk-token)
+      - [OP (OpenID Provider) Signature](#op-openid-provider-signature)
+      - [CIC (Client-Instance Claims) Signature](#cic-client-instance-claims-signature)
+      - [COS (Cosigner) Signature](#cos-cosigner-signature)
+  - [Types of PK Tokens](#types-of-pk-tokens)
+    - [Nonce-Commitment PK Token - Google Example](#nonce-commitment-pk-token---google-example)
+      - [Nonce-Commitment](#nonce-commitment)
+      - [MFA Cosigner Signature](#mfa-cosigner-signature)
+    - [GQ Signed Nonce-Commitment PK Tokens - Google Example](#gq-signed-nonce-commitment-pk-tokens---google-example)
+      - [GQ Signatures](#gq-signatures)
+      - [GQ Signature Protected Header](#gq-signature-protected-header)
+      - [GQ Signing](#gq-signing)
+    - [GQ Signed Audience-bound PK Tokens - GitHub Example](#gq-signed-audience-bound-pk-tokens---github-example)
+      - [Audience-Commitment](#audience-commitment)
+      - [GQ Signatures Are Required For Aud-Commitment PK Tokens](#gq-signatures-are-required-for-aud-commitment-pk-tokens)
+    - [GQ-Commitment PK Tokens (Gitlab-CI Example)](#gq-commitment-pk-tokens-gitlab-ci-example)
+      - [GQ-Commitment to the CIC](#gq-commitment-to-the-cic)
+      - [Security](#security)
+    - [ZK PK Tokens (Under development)](#zk-pk-tokens-under-development)
+  - [PK Token Compact Serialization](#pk-token-compact-serialization)
+    - [Refreshed Payload and Signature](#refreshed-payload-and-signature)
+- [Notes](#notes)
+  - [GQ Signatures](#gq-signatures-1)
+  - [Our JWS conventions](#our-jws-conventions)
```

Comment:
Similar feedback - may want to capitalize these headers to be consistent with the top portion